xbmc/cores/VideoPlayer/Edl.cpp

```diff
@@ -37,135 +40,129 @@ CEdl::CEdl()
-              __FUNCTION__, CURL::GetRedacted(strMovie));
+    CLog::LogF(LOGDEBUG,
+               "Checking for edit decision lists (EDL) on local drive or remote share for: {}",
+               CURL::GetRedacted(path));
 
-    /*
-     * Read any available file format until a valid EDL related file is found.
-     */
-    if (!bFound)
-      bFound = ReadVideoReDo(strMovie);
+    // Read any available file format until a valid EDL related file is found.
+    if (!found)
+      found = ReadVideoReDo(path);
 
-    if (!bFound)
-      bFound = ReadEdl(strMovie, fFramesPerSecond);
+    if (!found)
+      found = ReadEdl(path, fps);
 
-    if (!bFound)
-      bFound = ReadComskip(strMovie, fFramesPerSecond);
+    if (!found)
+      found = ReadComskip(path, fps);
 
-    if (!bFound)
-      bFound = ReadBeyondTV(strMovie);
+    if (!found)
+      found = ReadBeyondTV(path);
   }
   else
   {
-    bFound = ReadPvr(fileItem);
+    found = ReadPvr(fileItem);
   }
 
-  if (bFound)
+  if (found)
   {
     MergeShortCommBreaks();
     AddSceneMarkersAtStartAndEndOfEdits();
   }
 
-  return bFound;
+  return found;
 }
 
-bool CEdl::ReadEdl(const std::string& strMovie, const float fFramesPerSecond)
+bool CEdl::ReadEdl(const std::string& path, float fps)
 {
   Clear();
 
-  std::string edlFilename(URIUtils::ReplaceExtension(strMovie, ".edl"));
+  std::string edlFilename{URIUtils::ReplaceExtension(path, ".edl")};
   if (!CFile::Exists(edlFilename))
     return false;
 
   CFile edlFile;
   if (!edlFile.Open(edlFilename))
   {
-    CLog::Log(LOGERROR, "{} - Could not open EDL file: {}", __FUNCTION__,
-              CURL::GetRedacted(edlFilename));
+    CLog::LogF(LOGERROR, "Could not open EDL file: {}", CURL::GetRedacted(edlFilename));
     return false;
   }
 
-  bool bError = false;
-  int iLine = 0;
-  std::string strBuffer;
-  strBuffer.resize(1024);
-  while (edlFile.ReadString(&strBuffer[0], 1024))
+  bool error{false};
+  int line{0};
+  std::array<char, 1024> buffer{};
+  while (edlFile.ReadString(&buffer[0], 1024))
   {
     // Log any errors from previous run in the loop
-    if (bError)
-      CLog::Log(LOGWARNING, "{} - Error on line {} in EDL file: {}", __FUNCTION__, iLine,
-                CURL::GetRedacted(edlFilename));
+    if (error)
+      CLog::LogF(LOGWARNING, "Error on line {} in EDL file: {}", line,
+                 CURL::GetRedacted(edlFilename));
 
-    bError = false;
+    error = false;
+    line++;
 
-    iLine++;
+    std::array<char, 513> buffer1{};
+    std::array<char, 513> buffer2{};
+    int action{EDL::EDL_ACTION_NONE};
 
-    char buffer1[513];
-    char buffer2[513];
-    int iAction;
-    int iFieldsRead = sscanf(strBuffer.c_str(), "%512s %512s %i", buffer1,
-                             buffer2, &iAction);
-    if (iFieldsRead != 2 && iFieldsRead != 3) // Make sure we read the right number of fields
+    int fieldsRead =
+        sscanf(buffer.data(), "%512s %512s %i", buffer1.data(), buffer2.data(), &action);
```

Comment:
maybe an opportunity to use a stringstream instead of a c method here (for a future PR).